persist/fs/read.go

```diff
@@ -347,27 +349,33 @@ func (r *reader) Read() (ident.ID, checked.Bytes, uint32, error) {
 
 	n, err := r.dataReader.Read(data.Bytes())
 	if err != nil {
-		return none, nil, 0, err
+		return nil, nil, nil, 0, err
 	}
 	if n != int(entry.Size) {
-		return none, nil, 0, errReadNotExpectedSize
+		return nil, nil, nil, 0, errReadNotExpectedSize
 	}
 
-	r.entriesRead++
+	id := r.entryClonedID(entry.ID)
+	tags := r.entryClonedEncodedTagsTagIter(entry.EncodedTags)
 
-	return r.entryID(entry.ID), data, uint32(entry.Checksum), nil
+	r.entriesRead++
+	return id, tags, data, uint32(entry.Checksum), nil
 }
 
-func (r *reader) ReadMetadata() (id ident.ID, length int, checksum uint32, err error) {
-	var none ident.ID
+func (r *reader) ReadMetadata() (id ident.ID, tags ident.TagIterator, length int, checksum uint32, err error) {
 	if r.metadataRead >= r.entries {
-		return none, 0, 0, io.EOF
+		err = io.EOF
+		return
 	}
 
 	entry := r.indexEntriesByOffsetAsc[r.metadataRead]
+	id = r.entryClonedID(entry.ID)
+	tags = r.entryClonedEncodedTagsTagIter(entry.EncodedTags)
+	length = int(entry.Size)
+	checksum = uint32(entry.Checksum)
 
 	r.metadataRead++
-	return r.entryID(entry.ID), int(entry.Size), uint32(entry.Checksum), nil
+	return
```

Comment:
mind using `return id, tags, length, checksum, nil` instead? 
(IMO named returns hurt usability if used without this, cause it adds more things to consider when looking at exit points)

Reply:
I don't feel strongly, sure